src/server/components/charts-engine/components/processor/worker-chart-builder.ts

```diff
@@ -0,0 +1,180 @@
+import path from 'path';
+
+import workerpool from 'workerpool';
+
+import type {
+    DashWidgetConfig,
+    ServerChartsConfig,
+    Shared,
+    StringParams,
+} from '../../../../../shared';
+import {getServerFeatures} from '../../../../../shared';
+import {registry} from '../../../../registry';
+import type {WizardWorker} from '../worker/types';
+
+import {getChartApiContext} from './chart-api-context';
+import type {ChartBuilder} from './types';
+
+const pool = workerpool.pool(path.resolve(__dirname, '../worker'));
+async function getWorker(options: {timeout: number}) {
+    return pool
+        .proxy<WizardWorker>()
+        .timeout(options.timeout)
+        .catch((e) => {
+            pool.terminate();
+            throw e;
+        });
+}
+
+const ONE_SECOND = 1000;
+const JS_EXECUTION_TIMEOUT = ONE_SECOND * 9.5;
+
+type WizardChartBuilderArgs = {
+    userLogin: string | null;
+    userLang: string;
+    config: {
+        data: {
+            shared: string;
+        };
+    };
+    widgetConfig?: DashWidgetConfig['widgetConfig'];
+    isScreenshoter: boolean;
+};
+
+export const getWizardChartBuilder = async (
+    args: WizardChartBuilderArgs,
+): Promise<ChartBuilder> => {
+    const {config, widgetConfig, userLang} = args;
+    let shared: Record<string, any>;
+
+    const app = registry.getApp();
+    const features = getServerFeatures(app.nodekit.ctx);
+    const {getAvailablePalettesMap} = registry.common.functions.getAll();
+    const palettes = getAvailablePalettesMap();
+
+    // Nothing happens here - just for compatibility with the editor
+    const emptyStep =
+        (name: string) => async (options: {params: StringParams; actionParams: StringParams}) => {
+            const {params, actionParams} = options;
+            const timeStart = process.hrtime();
+            const context = getChartApiContext({
+                name,
+                shared,
+                params,
+                actionParams,
+                widgetConfig,
+                userLang,
+            });
+
+            return {
+                exports: {},
+                executionTiming: process.hrtime(timeStart),
+                name,
+                runtimeMetadata: context.__runtimeMetadata,
+            };
+        };
+
+    const chartBuilder: ChartBuilder = {
+        buildShared: async () => {
```

Comment:
Why did you use `async`?

Reply:
There is no specific reason
Just to match the [type](https://github.com/datalens-tech/datalens-ui/blob/f45b4b5b5b7e05e74b372e74c471ff974c1f768f/src/server/components/charts-engine/components/processor/types.ts#L141)